text/ProofModes.md

```diff
@@ -0,0 +1,118 @@
+# Proof Modes
+
+This CEP is about the infrastructure needed in order to have multiple 
+proof modes coexisting in a modular way (i.e. without syntactic or semantic conflicts)
+
+## Status quo (as per Coq 8.6)
+
+1. Tactic names are kernel names, i.e. unique, and can be accessed this way
+
+    ```
+    foo.tac arg1.
+    bar.tac arg1.
+    ```
+    
+1. Linking (not even requiring) a plugin loads into the *lexer* the tokens 
+   specific to the grammar implemented by the plugin
+   
+1. Linking (not even requiring) a plugin loads into the *parser* grammar the rules
+   specific to the plugin 
+   
```

Comment:
Well, the "official" API here is to use `freeze/unfreeze` if you are not happy about the changes, right? It is indeed very messy.